c++/src/capnp/schema.capnp

```diff
@@ -169,6 +169,36 @@ struct Node {
       targetsAnnotation @30 :Bool;
     }
   }
+
+  struct SourceInfo {
+    # Information gathered about the original source code from which this node was compiled.
+    #
+    # SourceInfo is delivered separately from the nodes themselves in order to make sure it is not
+    # accidentally included in cases where it is not needed. SourceInfo can be very big and is not
+    # generally used at runtime. Instead of educating everyone on how to remove it when not needed,
+    # we choose to deliver it out-of-band such that use cases that *do* need SourceInfo can request
+    # it explicitly. For example, `CodeGeneratorRequest` delivers a list of `SourceInfo`s in
+    # parallel with the list of `Node`s.
+
+    id @0 :UInt64;
+    # ID of the node we're describing.
+
+    docComment @1 :UInt64;
```

Comment:
... apparently because I was typing very quickly and there's no implementation to test against. :P